src/cli/CMakeLists.txt

```diff
@@ -13,22 +13,27 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-include_directories(../src)
+set(cli_SOURCES
+    Merge.cpp
+    Merge.h
+    EntropyMeter.cpp
+    EntropyMeter.h
+    Extract.cpp
+    Extract.h)
 
-add_executable(kdbx-extract kdbx-extract.cpp)
-target_link_libraries(kdbx-extract
-                      keepassx_core
-                      Qt5::Core
-                      ${GCRYPT_LIBRARIES}
-                      ${ZLIB_LIBRARIES})
+add_library(cli STATIC ${cli_SOURCES})
+target_link_libraries(cli Qt5::Core Qt5::Widgets)
 
-add_executable(kdbx-merge kdbx-merge.cpp)
-target_link_libraries(kdbx-merge
+add_executable(keepassxc-cli keepassxc-cli.cpp)
+target_link_libraries(keepassxc-cli
+                      cli
                       keepassx_core
                       Qt5::Core
                       ${GCRYPT_LIBRARIES}
-                      ${ZLIB_LIBRARIES})
-
+                      ${GPGERROR_LIBRARIES}
+                      ${ZLIB_LIBRARIES}
+                      zxcvbn)
 
-add_executable(entropy-meter entropy-meter.cpp)
-target_link_libraries(entropy-meter zxcvbn)
+install(TARGETS keepassxc-cli
+        BUNDLE DESTINATION . COMPONENT Runtime
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR} COMPONENT Runtime)
```

Comment:
You set the destination to `${BIN_INSTALL_DIR}` "declared" here https://github.com/keepassxreboot/keepassxc/blob/develop/CMakeLists.txt#L148
that is set to `.` and when executing `make install` goes to `/Application`

You can instead use 
```
include(GNUInstallDirs)
${CMAKE_INSTALL_BINDIR}
``` 
Or eventually hard code the `/usr/local/bin` path